File: app/src/main/java/eu/sesma/dagger/platform/AndroidApplication.java
package eu.sesma.dagger.platform;

import eu.sesma.dagger.di.ApplicationComponent;
import eu.sesma.dagger.di.DaggerApplicationComponent;

import android.app.Application;

public class AndroidApplication extends Application {

    private ApplicationComponent applicationComponent = createComponent();

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }

    protected ApplicationComponent createComponent() {
        return DaggerApplicationComponent.builder()
                .application(this)
                .build();
    }
}

File: app/src/main/java/eu/sesma/dagger/platform/AndroidApplication.kt
package eu.sesma.dagger.platform

import eu.sesma.dagger.di.ApplicationComponent
import eu.sesma.dagger.di.DaggerApplicationComponent

import android.app.Application

open class AndroidApplication : Application() {

    val applicationComponent = createComponent()

    protected open fun createComponent(): ApplicationComponent {
        return DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
    }
}
